internal/ir/xds.go

```diff
@@ -211,6 +212,9 @@ type HTTPRoute struct {
 	Destinations []*RouteDestination
 	// Rewrite to be changed for this route.
 	URLRewrite *URLRewrite
+	// RateLimit defines the more specific match conditions as well as limits for ratelimiting
+	// the requests on this route.
+	RateLimit *RateLimit
```

Comment:
@arkodg I don't understand why RateLimit doesn't use any of the RateLimitFilter fields. For example, #813 uses [JwtAuthenticationFilterProvider](https://github.com/envoyproxy/gateway/blob/main/api/v1alpha1/authenticationfilter_types.go#L61) instead of recreating the type in the IR. Since we own extension APIs, e.g. RateLimitFilter, we should leverage these types as much as possible.

Reply:
creating a decoupling from Front End as well as Back End API, that was the main goal of the IR, which is why we created custom structures even for known structures such as `StringMatch`

Reply:
as you mentioned, since the project owns the structure, should be safe to reuse it, I can update the PR with this change